api/types/access_list.go

```diff
@@ -0,0 +1,299 @@
+/*
+Copyright 2023 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package types
+
+import (
+	"encoding/json"
+	"fmt"
+	"time"
+
+	"github.com/gravitational/trace"
+
+	accesslistpb "github.com/gravitational/teleport/api/gen/proto/go/accesslist/v1"
+	"github.com/gravitational/teleport/api/types/header"
+	"github.com/gravitational/teleport/api/types/traits"
+	"github.com/gravitational/teleport/api/utils"
+)
+
+// FromAccessListV1 converts a v1 access list into an internal access list object.
+func FromAccessListV1(msg *accesslistpb.AccessList) (*AccessList, error) {
```

Comment:
nit: this should have test coverage.

Reply:
https://github.com/gravitational/teleport/pull/28787 Here's the PR where these tests are added in.